rocketpy/simulation/flight.py

```diff
@@ -1570,70 +1570,26 @@ def u_dot_generalized(self, t, u, post_processing=False):
         w = Vector([omega1, omega2, omega3])  # Angular velocity vector
 
         # Retrieve necessary quantities
-        rho = self.env.density.get_value_opt(z)
+        ## Rocket mass
         total_mass = self.rocket.total_mass.get_value_opt(t)
         total_mass_dot = self.rocket.total_mass_flow_rate.get_value_opt(t)
-        total_mass_ddot = self.rocket.total_mass_flow_rate.differentiate(t)
+        total_mass_ddot = self.rocket.total_mass_flow_rate.differentiate_complex_step(t)
         ## CM position vector and time derivatives relative to CDM in body frame
-        r_CM_z = (
-            -1
-            * (
-                (
-                    self.rocket.center_of_propellant_position
-                    - self.rocket.center_of_dry_mass_position
-                )
-                * self.rocket._csys
-            )
-            * self.rocket.motor.propellant_mass
-            / total_mass
-        )
-        r_CM = Vector([0, 0, r_CM_z.get_value_opt(t)])
-        r_CM_dot = Vector([0, 0, r_CM_z.differentiate(t)])
+        r_CM_z = self.rocket.z_coordinate_com_to_cdm
+        r_CM_t = r_CM_z.get_value_opt(t)
+        r_CM = Vector([0, 0, r_CM_t])
+        r_CM_dot = Vector([0, 0, r_CM_z.differentiate_complex_step(t)])
```

Comment:
Since we are pre-computing things, I wonder how different the execution time would be if we pre-calculated these derivatives with `Function.derivative_function`.

Then we would only have to interpolate inside `udot` instead of differentiating.
This would only work if the `Function` is discretized though (which I think everything insed udot is)

Reply:
Good suggestion, gonna work on this one very soon.

Reply:
Done! [7531ca6](https://github.com/RocketPy-Team/RocketPy/pull/595/commits/7531ca6a58f034613cf9554edc05031b8be130da)

Reply:
I did some changes and verified that the computational time required to run the flight simulation would increase (i.e. get worse) if we pre-calculate the derivative function.

```text
# before 1.38728 +- 0.22193
# after: 1.45481 +- 0.22315
```

We can better investigate it in the future.